libs/extensions/src/TTAlpsHistogramsFiller.cpp

```diff
@@ -112,4 +116,39 @@ void TTAlpsHistogramsFiller::FillHistograms1D(const std::shared_ptr<Event> event
       }
     }
   }
-}
\ No newline at end of file
+}
+
+void TTAlpsHistogramsFiller::FillLeadingPt(const std::shared_ptr<Event> event, std::string histName, std::vector<std::string> variableLocation) {
+  histogramsHandler->histograms1D[histName]->Fill(eventProcessor->GetMaxPt(event, variableLocation[0]));
+}
+
+void TTAlpsHistogramsFiller::FillSubLeadingPt(const std::shared_ptr<Event> event, std::string histName, std::vector<std::string> variableLocation) {
+  
+  float maxPt = eventProcessor->GetMaxPt(event, variableLocation[0]);
+  auto collection = event->GetCollection(variableLocation[0]);
+    for(auto object : *collection){
+      float pt = object->Get("pt");
+      if(pt == maxPt) continue;
+      histogramsHandler->histograms1D[histName]->Fill(pt);
+    }
+}
+
+void TTAlpsHistogramsFiller::FillCutFlowHist(const std::shared_ptr<CutFlowManager> cutFlowManager) {
+  int bin = 1;
+  int cutFlowLength = cutFlowManager->GetCutFlow().size();
+  TH1D* cutFlowHist = new TH1D("cutFlow", "cutFlow", cutFlowLength, 0, cutFlowLength+1);
+  for(auto &[name, weight] : cutFlowManager->GetCutFlow()){
+    cutFlowHist->SetBinContent(bin, weight);
+    cutFlowHist->GetXaxis()->SetBinLabel(bin, name.c_str());
+    bin++;
+  }
+  histogramsHandler->histograms1D["cutFlow"] = cutFlowHist;
+}
```

Comment:
I think the cut flow feeling is not ttalps specific - we should make it available for any analysis. Not sure what's the best way to achieve this, perhaps we should have a general HistogramFiller and its specializations for each analysis, similar to PhysicsObject.